### PR TITLE
Editorial for Line Segment Intersection

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC: https://open.kattis.com/submissions/14757118

### CODE REVIEW

File: solutions/platinum/kattis-segmentintersection.mdx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+source: Kattis
+title: Line Segment Intersection
+author: Rameez Parwez
+---
+
+[Unoffical Analysis](https://cp-algorithms.com/geometry/segments-intersection.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(T)$, where $T$ is the number of test cases.
```
Comment (SansPapyrus683) [inline solutions/platinum/kattis-segmentintersection.mdx:12]: **Time Complexity:** $\mathcal{O}(1)$ for each test case.

File: solutions/platinum/kattis-segmentintersection.mdx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+id: kattis-segmentintersection
+source: Kattis
+title: Line Segment Intersection
+author: Rameez Parwez
+---
+
+[Unoffical Analysis](https://cp-algorithms.com/geometry/segments-intersection.html)
+
+## Implementation
```
Comment (SansPapyrus683) [inline solutions/platinum/kattis-segmentintersection.mdx:10]: add a disclaimer that this is basically just the cp algo code spruced up

File: solutions/platinum/kattis-segmentintersection.mdx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+**Time Complexity:** $\mathcal{O}(T)$, where $T$ is the number of test cases.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// BeginCodeSnip{Finding intersection of two segments from CP Algorithms}
```
Comment (SansPapyrus683) [inline solutions/platinum/kattis-segmentintersection.mdx:21]:
```suggestion
// BeginCodeSnip{Line Segment Intersection (from link)}
```

File: solutions/platinum/kattis-segmentintersection.mdx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+		       betw(c.x, d.x, left.x) && betw(c.y, d.y, left.y);
+	}
+}
+// EndCodeSnip
+
+int main() {
+	int test_case;
+	cin >> test_case;
+	cout << fixed << setprecision(2);
+	for (int i = 1; i <= test_case; i++) {
```
Comment (SansPapyrus683) [inline solutions/platinum/kattis-segmentintersection.mdx:92]:
```suggestion
	int test_num;
	cin >> test_num;
	cout << fixed << setprecision(2);
	for (int t = 0; t < test_num; t++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/platinum/kattis-segmentintersection.mdx (hunk 1 above)
Comment (freakin23) [reply solutions/platinum/kattis-segmentintersection.mdx:12]: ok

File: solutions/platinum/kattis-segmentintersection.mdx (hunk 2 above)
Comment (freakin23) [reply solutions/platinum/kattis-segmentintersection.mdx:14]: done.

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by freakin23** [comment]
AC: https://open.kattis.com/submissions/14762548

**Pull request merged**